src/Adapter/Cache/CombineCompressCacheConfiguration.php

```diff
@@ -194,7 +194,7 @@ private function manageApacheOptimization($enabled)
 
         // feature activation
         if (false === $isCurrentlyEnabled && true === $enabled) {
-            if ($this->tools->generateHtaccess()) {
+            if ($this->tools->generateHtaccess(null, null, 1)) {
                 $this->configuration->set('PS_HTACCESS_CACHE_CONTROL', true);
```

Comment:
Maybe the configuration should be set before calling `generateHtaccess()`? That way we wouldn't need to pass any parameters.

Reply:
That would simplify the change but the user won't have any kind of feedback if the operation fails.

Reply:
@eternoendless

Reply:
Why would that be? You just need to tweak the code a little like this:

```php
$this->configuration->set('PS_HTACCESS_CACHE_CONTROL', true);
if (!$this->tools->generateHtaccess()) {
    $errors = array(
       // ...
```

Reply:
@eternoendless Thank you for the feedback. I have updated the code